services/web-app/pages/_middleware.js

```diff
@@ -32,12 +32,26 @@ function exitWith404(req) {
 }
 
 export async function middleware(req) {
+  const protocol =
+    getRunMode === RunMode.Prod || process.env.RUNNING_SECURELY === '1' ? 'https' : 'http'
+
+  // log page request
+  fetch(`${protocol}://localhost:${req.nextUrl.port}/api/log-message`, {
```

Comment:
Do you think we should have a helper function outside of this middleware which takes a request and generates a loopback fetch URL?

Reply:
Or maybe a wrapper around sending loopback requests altogether. Basically something that would handle all of the stuff we need to hand-write for each of these fetch requests.

Reply:
added services/web-app/utils/sendLocalRequest.js